### PR TITLE
Implement systemtags through webdav

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Exposes system tags under the `{http://nextcloud.org/ns}system-tags` property.

### CODE REVIEW

**Comment by icewind1991** [comment]
https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/apps/dav/lib/Connector/Sabre/TagsPlugin.php might provide the info you need already

**Comment by tobiasKaminsky** [comment]
Strange, it does always return empty, but not 404:
```
	<d:response>
		<d:href>/remote.php/dav/files/tobias/temp/tags/1.md</d:href>
		<d:propstat>
			<d:prop>
				<oc:id>07264386ocmk9xspynvl</oc:id>
				<oc:fileid>7264386</oc:fileid>
				<d:displayname>1.md</d:displayname>
				<d:resourcetype/>
				<oc:favorite>0</oc:favorite>
				<nc:mount-type></nc:mount-type>
				<d:creationdate>1970-01-01T00:00:00+00:00</d:creationdate>
				<d:getetag>&quot;643647e3a10a7&quot;</d:getetag>
				<oc:permissions>RGDNVW</oc:permissions>
				<oc:share-types/>
				<oc:owner-id>tobias</oc:owner-id>
				<d:getcontenttype>text/markdown</d:getcontenttype>
				<nc:sharees/>
				<oc:tags/>
			</d:prop>
			<d:status>HTTP/1.1 200 OK</d:status>
		</d:propstat>
		<d:propstat>
			<d:prop>
				<nc:is-encrypted/>
				<nc:share-expiration/>
				<nc:rich-workspace/>
				<nc:X-Hash-MD5/>
				<d:quota-used-bytes/>
			</d:prop>
			<d:status>HTTP/1.1 404 Not Found</d:status>
		</d:propstat>
	</d:response>
```

Double tested, and also with our c.nc.c.

Request is:
```
curl --request PROPFIND \
  --url https://cloud.nextcloud.com/remote.php/dav/files/tobias/temp/tags/ \
  --header 'Content-Type: application/xml' \
  --data '<?xml version="1.0" encoding="UTF-8"?>
<D:propfind xmlns:D="DAV:">
<D:prop>
<id xmlns="http://owncloud.org/ns"/>
<fileid xmlns="http://owncloud.org/ns"/>
<D:displayname/>
<D:resourcetype/>
<favorite xmlns="http://owncloud.org/ns"/>
<is-encrypted xmlns="http://nextcloud.org/ns"/>
<mount-type xmlns="http://nextcloud.org/ns"/>
<D:creationdate/>
<id xmlns="http://owncloud.org/ns"/>
<D:getetag/>
<permissions xmlns="http://owncloud.org/ns"/>
<share-types xmlns="http://owncloud.org/ns"/>
<share-expiration xmlns="http://nextcloud.org/ns"/>
<rich-workspace xmlns="http://nextcloud.org/ns"/>
<owner-id xmlns="http://owncloud.org/ns"/>
	<X-Hash-MD5 xmlns="http://nextcloud.org/ns"/>
	<D:quota-used-bytes/>
<D:getcontenttype/>
<sharees  xmlns="http://nextcloud.org/ns"/>
<tags xmlns="http://owncloud.org/ns"/>
</D:prop></D:propfind>'
```

**Comment by icewind1991** [comment]
Looks like that dav property return the user-specific tags (not sure if they are used for anything but favorites atm) instead of system tags.

**Comment by icewind1991** [comment]
Implemented showing system tags with the `{http://nextcloud.org/ns}system-tags` prop

**Comment by skjnldsv** [comment]
> not sure if they are used for anything but favorites atm

they are not

**Comment by tobiasKaminsky** [comment]
Many thanks!
This works!

**Comment by tobiasKaminsky** [comment]
```
<nc:system-tag oc:can-assign="true" oc:id="3" oc:user-assignable="true" oc:user-visible="true">important</nc:system-tag>
```
can it happen that this webdav request will return an entry with: user-visible=false?
Do clients have to check this?

**Comment by icewind1991** [comment]
No, those should be filtered out server side

**Comment by tobiasKaminsky** [comment]
Can we merge this soon, as I need it to have proper tests on Android.

**Comment by skjnldsv** [comment]
Fixing psalm and we're good to go

**Review by skjnldsv — APPROVED** [review]

File: apps/dav/lib/SystemTag/SystemTagList.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * @copyright Copyright (c) 2016, ownCloud, Inc.
```
Comment (blizzz) [inline apps/dav/lib/SystemTag/SystemTagList.php:3]: *cough*

**Comment by blizzz** [comment]
There are some similarities to https://github.com/nextcloud/server/pull/37961, but also differences (like depth is respected here, and we use the search there, plus get more meta data)

File: apps/dav/lib/SystemTag/SystemTagList.php (hunk 1 above)
Comment (skjnldsv) [reply apps/dav/lib/SystemTag/SystemTagList.php:3]: Fixing it

**Review by blizzz — APPROVED** [review]

**Comment by skjnldsv** [comment]
Hum, phpunit is failing

```

**Pull request merged**